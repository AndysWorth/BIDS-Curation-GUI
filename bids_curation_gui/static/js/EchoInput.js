import React from 'react';
import PropTypes from 'prop-types';

/**
 * Text box component with a submit button.
 *
 * @param {String} value Current value of the text box
 * @param {Function} onChange Change handler for text input
 * @param {Function} onSubmit Form submission handler
 */
const EchoInput = ({value, onChange, onSubmit}) => (
  <form onSubmit={onSubmit}>
    <div className="form-row">
      <div className="form-group col-sm-11">
        <input autoFocus type="text" className="form-control" placeholder="Enter something" value={value} onChange={onChange}/>
      </div>
      <div className="col-sm-1">
        <button className="btn btn-primary" type="submit">Submit</button>
      </div>
    </div>
  </form>
);

EchoInput.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};

export default EchoInput;