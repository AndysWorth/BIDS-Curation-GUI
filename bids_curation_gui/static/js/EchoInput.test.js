import React from 'react';
import { shallow } from 'enzyme';
import EchoInput from './EchoInput';

describe('EchoInput', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            value: 'test-value',
            onChange: jest.fn(),
            onSubmit: jest.fn(),
        };
        wrapper = shallow(<EchoInput {...props} />);
    });

    it('should render an initial value', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a new value when updated', () => {
        wrapper.setProps({value: 'new-test-value'});
        expect(wrapper).toMatchSnapshot();
    });

    it('should invoke onSubmit when the submit button is clicked', () => {
        wrapper.find('form').simulate('submit');
        expect(props.onSubmit).toHaveBeenCalled();
    });

    it('should invoke onChange when the input text changes', () => {
        wrapper.find('input').simulate('change');
        expect(props.onChange).toHaveBeenCalled();
    });
});

// test('EchoInput renders values on change', () => {
//     const
//     const onSubmit = jest.fn();

//     const component = renderer.create(

//     );


//     // Change the value property
//     component.setState('value', 'new-value');
//     tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//     expect(onChange.mock.calls.length).toBe(0);

//     // Submit the form
//     tree.props.onSubmit();
//     expect(tree).toMatchSnapshot();
//     expect(onSubmit.mock.calls.length).toBe(1);
// });

// test('EchoInput submits on click', () => {

// });