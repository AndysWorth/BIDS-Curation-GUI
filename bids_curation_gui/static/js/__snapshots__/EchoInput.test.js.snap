// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EchoInput should render a new value when updated 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div
    className="form-row"
  >
    <div
      className="form-group col-sm-11"
    >
      <input
        autoFocus={true}
        className="form-control"
        onChange={[MockFunction]}
        placeholder="Enter something"
        type="text"
        value="new-test-value"
      />
    </div>
    <div
      className="col-sm-1"
    >
      <button
        className="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
</form>
`;

exports[`EchoInput should render an initial value 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div
    className="form-row"
  >
    <div
      className="form-group col-sm-11"
    >
      <input
        autoFocus={true}
        className="form-control"
        onChange={[MockFunction]}
        placeholder="Enter something"
        type="text"
        value="test-value"
      />
    </div>
    <div
      className="col-sm-1"
    >
      <button
        className="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </div>
  </div>
</form>
`;
